spec/models/event_with_keyword_arguments.rb

```diff
@@ -9,8 +9,16 @@ class EventWithKeywordArguments
       before :_before_close
       transitions from: :open, to: :closed
     end
+
+    event :another_close do
+      before :_before_another_close
+      transitions from: :open, to: :closed
+    end
   end
 
   def _before_close(key:)
   end
```

Comment:
Can we please include a scenario where key is optional?
```
[1] pry(main)> def method_a(key:)
[1] pry(main)* end
=> :method_a
[2] pry(main)> def method_b(key: nil)
[2] pry(main)* end
=> :method_b
[4] pry(main)> method(:method_a).arity
=> 1
[5] pry(main)> method(:method_b).arity
=> -1
[6] pry(main)>
```

That should fall within `invoke_with_variable_arity` which from my understanding will fail.

Reply:
My solution no longer relies on checking for arity, so it might still work.

I'll do this when I have time. If you want feel free to create a PR on my fork 💯